crates/nu-command/src/bytes/index_of.rs

```diff
@@ -0,0 +1,142 @@
+use super::{operate, BytesArgument};
+use nu_engine::CallExt;
+use nu_protocol::ast::{Call, CellPath};
+use nu_protocol::engine::{Command, EngineState, Stack};
+use nu_protocol::{
+    Category, Example, PipelineData, ShellError, Signature, Span, SyntaxShape, Value,
+};
+
+struct Arguments {
+    pattern: Vec<u8>,
+    end: bool,
+    column_paths: Option<Vec<CellPath>>,
+}
+
+impl BytesArgument for Arguments {
+    fn take_column_paths(&mut self) -> Option<Vec<CellPath>> {
+        self.column_paths.take()
+    }
+}
+
+#[derive(Clone)]
+pub struct BytesIndexOf;
+
+impl Command for BytesIndexOf {
+    fn name(&self) -> &str {
+        "bytes index-of"
+    }
+
+    fn signature(&self) -> Signature {
```

Comment:
I'm wondering if we should have an `--all` parameter that searched the entire binary input that is passed to it. This would be super helpful for finding multi-page offsets  in TIFF files or TAR files. For instance, if every TIFF page in a multi-page TIFF starts with 0x[123] and then we return all of them, it would then be easy to store that list/array and iterate through it, doing something to each item.

Reply:
Ah, thank you, that's a good suggestion, I'll try it

Reply:
Sorry, I have a simple question, for the following case
```
0x[11 11 11 11] | bytes index-of 0x[11 11]
```

Should we return:
1. [0 1 2], or
2. [0 2]?

Reply:
I think the offsets should be 0 and 2 for [11 11], once we search a range I don't think we should search it again.

Reply:
Ok, that make sence :)